sdk/storage/Azure.Storage.Blobs/src/BlockBlobWriteStream.cs

```diff
@@ -0,0 +1,99 @@
+        private readonly BlobRequestConditions _conditions;
+        private readonly List<string> _blockIds;
+
+        public BlockBlobWriteStream(
+            BlockBlobClient blockBlobClient,
+            long bufferSize,
+            long position,
+            BlobRequestConditions conditions,
+            IProgress<long> progressHandler) : base(
+                position,
+                bufferSize,
+                progressHandler)
+        {
+            ValidateBufferSize(bufferSize);
+            _blockBlobClient = blockBlobClient;
+            _conditions = conditions;
+            _blockIds = new List<string>();
+        }
+
+        protected override async Task AppendInternal(bool async, CancellationToken cancellationToken)
+        {
+            if (_buffer.Length > 0)
+            {
+                _buffer.Position = 0;
+
+                string blockId = StorageExtensions.GenerateBlockId(_position + _buffer.Length);
+
+                if (async)
+                {
+                    await _blockBlobClient.StageBlockAsync(
+                        base64BlockId: blockId,
+                        content: _buffer,
+                        conditions: _conditions,
+                        progressHandler: _progressHandler,
+                        cancellationToken: cancellationToken)
+                        .ConfigureAwait(false);
+                }
+                else
+                {
+                    _blockBlobClient.StageBlock(
+                        base64BlockId: blockId,
+                        content: _buffer,
+                        conditions: _conditions,
+                        progressHandler: _progressHandler,
+                        cancellationToken: cancellationToken);
+                }
+
+                _blockIds.Add(blockId);
+                _buffer.Clear();
+            }
+        }
+
+        protected override async Task FlushInternal(bool async, CancellationToken cancellationToken)
+        {
+            await AppendInternal(async, cancellationToken).ConfigureAwait(false);
+
+            await _blockBlobClient.CommitBlockListInternal(
+                base64BlockIds: _blockIds,
+                blobHttpHeaders: default,
+                metadata: default,
+                tags: default,
+                conditions: _conditions,
+                accessTier: default,
+                async: async,
+                cancellationToken: cancellationToken)
+                .ConfigureAwait(false);
+        }
+
+        protected override void ValidateBufferSize(long bufferSize)
+        {
+            if (bufferSize < 1)
+            {
+                throw new ArgumentOutOfRangeException(nameof(bufferSize), "Must be >= 1");
+            }
+
+            if (bufferSize > Constants.Blob.Block.MaxStageBytes)
+            {
+                throw new ArgumentOutOfRangeException(nameof(bufferSize), $"Must <= {Constants.Blob.Block.MaxStageBytes}");
```

Comment:
Nit - `Must be less than or equal to {...}`

Reply:
Fixed in next iteration.